com.hazelcast.jet.core.JobTimeoutTest.when_jobIsResumedAndExceedsTimeout_jobIsCancelled [HZ-1707]

_5.2.z_ (commit 95533811f23cb7d42f38f6c8fbeb87b577352aec)

Failed on Openj9-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-Openj9-17/35/testReport/com.hazelcast.jet.core/JobTimeoutTest/when_jobIsResumedAndExceedsTimeout_jobIsCancelled/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: jobId=08dd-1441-6e40-0001 expected:<SUSPENDED> but was:<COMPLETING>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at com.hazelcast.jet.core.JetTestSupport.lambda$assertJobStatusEventually$2(JetTestSupport.java:305)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1253)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:304)
	at com.hazelcast.jet.core.JobTimeoutTest.when_jobIsResumedAndExceedsTimeout_jobIsCancelled(JobTimeoutTest.java:94)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:858)

```
</details>
<details><summary>Standard output:</summary>

```
10:42:02,289  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.2.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
10:42:02,290  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
10:42:02,290  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
10:42:02,290  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Hazelcast Platform 5.2.1-SNAPSHOT (20221029 - 9553381) starting at [127.0.0.1]:5701
10:42:02,290  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Cluster name: dev
10:42:02,290  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
10:42:02,290  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Jet is enabled
10:42:02,291  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
10:42:02,294  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
10:42:02,295  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
10:42:02,296  INFO || - [Diagnostics] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
10:42:02,296  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] [127.0.0.1]:5701 is STARTING
10:42:02,296  INFO || - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 57ef8bea-ce21-49d6-a7fd-f9aad48d8503 this
]

10:42:02,296  INFO || - [JobCoordinationService] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Jet started scanning for jobs
10:42:02,296 DEBUG || - [JobCoordinationService] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
10:42:02,297  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] [127.0.0.1]:5701 is STARTED
10:42:02,298  INFO || - [PartitionStateManager] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Initializing cluster partition table arrangement...
10:42:02,299 DEBUG || - [JobClassLoaderService] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Creating job classLoader for job 08dd-1441-6e40-0001
10:42:02,396 DEBUG || - [JobRepository] hz.magical_elgamal.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Job cleanup took 0ms
10:42:02,417 DEBUG || - [JobClassLoaderService] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Create processor classloader map for job 08dd-1441-6e40-0001
10:42:02,419  INFO || - [JobCoordinationService] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Starting job 08dd-1441-6e40-0001 based on submit request
10:42:02,421  INFO || - [MasterJobContext] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Didn't find any snapshot to restore for job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001
10:42:02,421  INFO || - [MasterJobContext] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Start executing job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001, execution graph in DOT format:
digraph DAG {
	"stuck" [localParallelism=2];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
10:42:02,421 DEBUG || - [MasterJobContext] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Building execution plan for job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001
10:42:02,421 DEBUG || - [MasterJobContext] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Built execution plans for job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001
10:42:02,422 DEBUG || - [InitExecutionOperation] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Initializing execution plan for job 08dd-1441-6e40-0001, execution 08dd-1441-6e41-0001 from [127.0.0.1]:5701
10:42:02,424  INFO || - [JobExecutionService] hz.magical_elgamal.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Execution plan for jobId=08dd-1441-6e40-0001, jobName='08dd-1441-6e40-0001', executionId=08dd-1441-6e41-0001 initialized
10:42:02,425 DEBUG || - [MasterJobContext] hz.magical_elgamal.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Init of job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001 was successful
10:42:02,425 DEBUG || - [MasterJobContext] hz.magical_elgamal.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Executing job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001
10:42:02,425  INFO || - [JobExecutionService] hz.magical_elgamal.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Start execution of job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001 from coordinator [127.0.0.1]:5701
10:42:02,426  INFO || - [HazelcastTestSupport] hz.magical_elgamal.jet.blocking.thread-0 - MockP.init called on hz.magical_elgamal.jet.blocking.thread-0
10:42:02,426  INFO || - [HazelcastTestSupport] hz.magical_elgamal.jet.blocking.thread-1 - MockP.init called on hz.magical_elgamal.jet.blocking.thread-1
10:42:02,621 DEBUG || - [Job] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Sending SUSPEND_GRACEFUL request for job 08dd-1441-6e40-0001 (name ??)
10:42:03,423  WARN || - [JobCoordinationService] hz.magical_elgamal.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] null
java.lang.IllegalStateException: Cannot CANCEL_FORCEFUL: Job is already terminating in mode: SUSPEND_FORCEFUL
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$terminateJob$14(JobCoordinationService.java:492) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$toNullFunction$45(JobCoordinationService.java:869) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$46(JobCoordinationService.java:898) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$55(JobCoordinationService.java:1327) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:858) ~[?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
10:42:03,549  INFO || - [HazelcastTestSupport] ForkJoinPool.commonPool-worker-258 - MockP.close called on ForkJoinPool.commonPool-worker-258
10:42:03,549  INFO || - [HazelcastTestSupport] ForkJoinPool.commonPool-worker-258 - MockP.close called on ForkJoinPool.commonPool-worker-258
10:42:03,549 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-258 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Completed execution of job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001
10:42:03,549 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-258 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Execution of job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001 completed with failure
java.util.concurrent.CompletionException: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: SUSPEND_FORCEFUL
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162) ~[?:?]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:72) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$ExecutionTracker.taskletDone(TaskletExecutionService.java:498) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:330) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:858) ~[?:?]
Caused by: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: SUSPEND_FORCEFUL
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:318) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:722) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution(JobExecutionService.java:718) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.TerminateExecutionOperation.doRun(TerminateExecutionOperation.java:59) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:55) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:190) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:283) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:258) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:219) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:601) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:580) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:541) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:241) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:61) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:294) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:277) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$cancelExecutionInvocations$20(MasterJobContext.java:645) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:858) ~[?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
10:42:03,550 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-258 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001 received response to StartExecutionOperation from [127.0.0.1]:5701: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: SUSPEND_FORCEFUL
10:42:03,550 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-258 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Execution of job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001 has failures: [[127.0.0.1]:5701=com.hazelcast.jet.impl.exception.JobTerminateRequestedException: SUSPEND_FORCEFUL]
10:42:03,551 DEBUG || - [JobClassLoaderService] hz.magical_elgamal.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=08dd-1441-6e40-0001
10:42:03,625 DEBUG || - [AbstractJobProxy] Time-limited test - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Sending CANCEL_FORCEFUL request for job 08dd-1441-6e40-0001 (name ??)
10:42:03,625  WARN || - [JoinSubmittedJobOperation] hz.magical_elgamal.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.impl.MasterJobContext.requestTermination(MasterJobContext.java:384) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$terminateJob$14(JobCoordinationService.java:490) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$toNullFunction$45(JobCoordinationService.java:869) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$46(JobCoordinationService.java:898) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$55(JobCoordinationService.java:1327) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:858) ~[?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
10:42:03,626 DEBUG || - [JobCoordinationService] hz.magical_elgamal.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] job '08dd-1441-6e40-0001', execution 08dd-1441-6e41-0001 is completed
10:42:03,627  WARN || - [JoinSubmittedJobOperation] ForkJoinPool.commonPool-worker-256 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] at com.hazelcast.jet.impl.MasterJobContext.requestTermination(MasterJobContext.java:397)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$terminateJob$14(JobCoordinationService.java:490)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$toNullFunction$45(JobCoordinationService.java:869)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$46(JobCoordinationService.java:898)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$55(JobCoordinationService.java:1327)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:858)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

java.util.concurrent.CancellationException: at com.hazelcast.jet.impl.MasterJobContext.requestTermination(MasterJobContext.java:397)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$terminateJob$14(JobCoordinationService.java:490)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$toNullFunction$45(JobCoordinationService.java:869)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$46(JobCoordinationService.java:898)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$55(JobCoordinationService.java:1327)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:858)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

	at com.hazelcast.jet.impl.JobResult.getFailureAsThrowable(JobResult.java:104) ~[classes/:?]
	at com.hazelcast.jet.impl.JobResult.asCompletableFuture(JobResult.java:118) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$46(JobCoordinationService.java:904) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$55(JobCoordinationService.java:1327) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:858) ~[?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
10:42:03,627  INFO || - [JetTestSupport] Time-limited test - Terminating instanceFactory in JetTestSupport.@After
10:42:03,628  INFO || - [LifecycleService] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
10:42:03,628  WARN || - [Node] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Terminating forcefully...
10:42:03,628  INFO || - [Node] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Shutting down connection manager...
10:42:03,628  INFO || - [Node] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Shutting down node engine...
10:42:03,630  INFO || - [NodeExtension] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Destroying node NodeExtension.
10:42:03,630  INFO || - [Node] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] Hazelcast Shutdown is completed in 2 ms.
10:42:03,630  INFO || - [LifecycleService] Thread-34375 - [127.0.0.1]:5701 [dev] [5.2.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after when_jobIsResumedAndExceedsTimeout_jobIsCancelled(com.hazelcast.jet.core.JobTimeoutTest): BuildInfo{version='5.2.1-SNAPSHOT', build='20221029', buildNumber=20221029, revision=9553381, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'when_jobIsResumedAndExceedsTimeout_jobIsCancelled(com.hazelcast.jet.core.JobTimeoutTest):'
10:42:00, accumulated pauses: 118 ms, max pause: 33 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/JobTimeoutTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/JobTimeoutTest.java
@@ -106,10 +106,10 @@ public class JobTimeoutTest extends JetTestSupport {
         final JobConfig jobConfig = new JobConfig().setTimeoutMillis(1000L);
         final Job job = hz.getJet().newJob(dag, jobConfig);
 
-        assertJobStatusEventually(job, JobStatus.RUNNING, 1);
+        assertJobStatusEventually(job, JobStatus.RUNNING);
         job.suspend();
 
-        assertJobStatusEventually(job, JobStatus.SUSPENDED, 1);
+        assertJobStatusEventually(job, JobStatus.SUSPENDED);
 
         assertThrows(CancellationException.class, job::join);
         assertEquals(JobStatus.FAILED, job.getStatus());

--- a/hazelcast/src/test/java/com/hazelcast/jet/core/JobTimeoutTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/core/JobTimeoutTest.java
@@ -88,10 +88,10 @@ public class JobTimeoutTest extends JetTestSupport {
         final JobConfig jobConfig = new JobConfig().setTimeoutMillis(1000L);
         final Job job = hz.getJet().newJob(dag, jobConfig);
 
-        assertJobStatusEventually(job, JobStatus.RUNNING, 1);
+        assertJobStatusEventually(job, JobStatus.RUNNING);
         job.suspend();
 
-        assertJobStatusEventually(job, JobStatus.SUSPENDED, 1);
+        assertJobStatusEventually(job, JobStatus.SUSPENDED);
         job.resume();
 
         assertThrows(CancellationException.class, job::join);

